From d3edd2ec659186038ba5c1df912008c5e4c59288 Mon Sep 17 00:00:00 2001
From: Bastian Cieslik <bastiancieslik@web.de>
Date: Sun, 22 Mar 2026 10:33:34 +0100
Subject: [PATCH] drv: spi: spidev Add orangepi spi-dev compatible

Original patch of commit (0c5e69360b359ba1331bf06e9f200ac716d0b832) no longer working.
refer to: https://github.com/orangepi-xunlong/linux-orangepi/commit/0c5e69360b359ba1331bf06e9f200ac716d0b832
---
 drivers/spi/spidev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index d8f056890..7f9fecf5e 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -691,6 +691,7 @@ static const struct file_operations spidev_fops = {
 static struct class *spidev_class;
 
 static const struct spi_device_id spidev_spi_ids[] = {
+    { .name = "spi-dev" },
 	{ .name = "bh2228fv" },
 	{ .name = "dh2228fv" },
 	{ .name = "jg10309-01" },
@@ -720,6 +721,7 @@ static int spidev_of_check(struct device *dev)
 }
 
 static const struct of_device_id spidev_dt_ids[] = {
+    { .compatible = "orangepi,spi-dev", .data = &spidev_of_check },
 	{ .compatible = "cisco,spi-petra", .data = &spidev_of_check },
 	{ .compatible = "dh,dhcom-board", .data = &spidev_of_check },
 	{ .compatible = "elgin,jg10309-01", .data = &spidev_of_check },
--
2.47.3

